dev/io.openliberty.microprofile.openapi.internal.common/src/io/openliberty/microprofile/openapi/internal/common/OpenAPIEndpointManager.java

```diff
@@ -0,0 +1,172 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ******************************************************************************/
+
+package io.openliberty.microprofile.openapi.internal.common;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.kernel.productinfo.ProductInfo;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.*;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Component(configurationPolicy = ConfigurationPolicy.OPTIONAL, immediate = true, configurationPid = "io.openliberty.microprofile.openapi")
+public class OpenAPIEndpointManager {
+
+    private static final TraceComponent tc = Tr.register(OpenAPIEndpointManager.class);
+
+    private static final String OPEN_API_DOC_VAR_NAME = "openAPIDOCPATH";
+    private static final String OPEN_API_DOC_ENDPOINT_CONFIG_NAME = "docPath";
+    private static final String OPEN_API_DOC_ENDPOINT_PATH = "/openapi";
+
+    private static final String OPEN_API_UI_VAR_NAME = "openAPIUIPATH";
+    private static final String OPEN_API_UI_CONFIG_NAME = "uiPath";
+    private static final String OPEN_API_UI_PATH = "/ui";
+
+    private static final Pattern PATH_PATTERN = Pattern.compile("^(/[\\w./_-]*)?$");
+    private static final Map<String, String> CONFLICTING_PATHS_MAP = Stream.of(
+            new AbstractMap.SimpleEntry<>("/ibm/api", "OpenAPI"),
+            new AbstractMap.SimpleEntry<>("/health", "mpHealth"),
+            new AbstractMap.SimpleEntry<>("/metrics", "mpMetrics")).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+    private String uiPath = null;
+    private String docPath = null;
+
+    private WABConfigManager UIWabConfigManager;
+    private WABConfigManager DocWabConfigManager;
+
+    @Activate
+    @Modified
+    protected void activate(BundleContext context, Map<String, Object> properties) {
+        docPath = OPEN_API_DOC_ENDPOINT_PATH;
+        uiPath = docPath + OPEN_API_UI_PATH;
+
+        if (ProductInfo.getBetaEdition()) {
+            //check for system property `open_api_path_enabled` as additional guide - getBoolean returns `true` if the value exists and is set to `true`, if the value is `false`
+            if (Boolean.getBoolean("open_api_path_enabled")) {
+                resolvePathFromProperties(properties);
+            }
+        }
+        UIWabConfigManager = new WABConfigManager(context, OPEN_API_UI_VAR_NAME, uiPath, "OpenAPI UI");
+        DocWabConfigManager = new WABConfigManager(context, OPEN_API_DOC_VAR_NAME, docPath, "OpenAPI Doc");
+        UIWabConfigManager.register();
+        DocWabConfigManager.register();
```

Comment:
If this is called in response to a config modification, we need to unregister the old config managers (or update them rather than creating new ones).